app.py

```diff
@@ -1,24 +1,131 @@
-from flask import url_for,render_template,redirect,Flask,flash
-from forms import LoginForm
+from flask import Flask, render_template, flash, redirect, url_for, session, request, logging, jsonify
+from passlib.hash import sha256_crypt
+from forms import RegisterForm, LoginForm
 from Make_Tables.mysqlconnect import mydb, mycursor, create_insert_statement   #Imported the mysqlconnect.py file from Make_tables folder
+import gc
+from helper import not_logged_in, is_logged_in
+
 
 app=Flask(__name__,static_url_path='/public')
 app.config['SECRET_KEY']='c828b6ff21f45063fd7860e5c1b1d233'
+app.config['MYSQL_CURSORCLASS'] = 'DictCursor'
+
 
+# Home page before logging in
 @app.route('/')
 def home():
-    return render_template('index.html')
-
-@app.route("/login",methods=['GET','POST'])
-def Login():
-    form=LoginForm()
-    if form.validate_on_submit():
-        if form.email.data[1:6].isdecimal() and form.email.data[6:]=='@students.iitmandi.ac.in' :
-            flash('You have logged-in successfully!')
-            return redirect(url_for('home'))
+    return 'Home page of SAC-Portal.'
+
+
+# User Registration
+@app.route('/register', methods=['GET', 'POST'])
+@not_logged_in
+def register():
+
+    form = RegisterForm(request.form)
+
+    if request.method == 'POST' and form.validate():
+        
+        if (form.userID.data[0] == 'b' or form.userID.data[0] == 'd' or form.userID.data[0] == 't' or form.userID.data[0] == 's' or form.userID.data[0] == 'v' ) and form.userID.data[1:6].isdecimal :
+            userID = form.userID.data
+        else:
+            return 'Not a valid userID!'
+
+        if form.email.data[0:6] == form.userID.data and form.email.data[6:]=='@students.iitmandi.ac.in' :
+            email = form.email.data
         else:
-            flash('Login Unsuccessful. Invalid Email/Password')
-    return render_template('login.html',title='Login | SAC Portal, IIT Mandi',form=form)
+            return 'Please enter valid email!'
+        name = form.name.data
+        password = sha256_crypt.hash(str(form.password.data))
+        admin = 0
+        
+        mycursor.execute("SELECT * FROM Users WHERE userID = (%s)", (userID,))
+
+        row = mycursor.fetchone()
+
+        if row != None:   #checking for duplicate Users(if User already registered, can't register again)
+            return 'This User already exists!'
+        else:
+
+            # Execute query
+            mycursor.execute("INSERT INTO Users(userID, email, password, name, admin) VALUES(%s, %s, %s, %s, %s)", (userID, email, password, name, admin))
```

Comment:
- [ ]  check and make sure these variables have no unesscessary special characters, eg. name = "Milind; DROP TABLE USERS;"

Reply:
Does this mean SQL Injections?

Reply:
Yeah , I guess this has to be done to handle SQL injections